ctapipe/io/eventfilereader.py

```diff
@@ -382,19 +231,25 @@ class EventFileReaderFactory(Factory):
     subclasses = Factory.child_subclasses(EventFileReader)
     subclass_names = [c.__name__ for c in subclasses]
 
-    reader = CaselessStrEnum(subclass_names, None, allow_none=True,
-                             help='Event file reader to use. If None then '
-                                  'a reader will be chosen based on file '
-                                  'extension').tag(config=True)
+    reader = CaselessStrEnum(
+        subclass_names,
+        None,
+        allow_none=True,
+        help='Event file reader to use. If None then a reader will be chosen '
+             'based on file extension'
+    ).tag(config=True)
 
     # Product classes traits
     # Would be nice to have these automatically set...!
-    input_path = Unicode(get_dataset('gamma_test.simtel.gz'), allow_none=True,
-                         help='Path to the input file containing '
-                              'events.').tag(config=True)
-    max_events = Int(None, allow_none=True,
-                     help='Maximum number of events that will be read from'
-                          'the file').tag(config=True)
+    input_path = Unicode(
```

Comment:
I still disagree that this is a trait. I think this should be an argument to `__init__`. A `Tool` can have a trait `input_path` to make it available as config or commandline option but it should then pass it to `EventFileReader.__init__` or EventFileReaderFactory.__init__`. This is not configuration, this is an positional argument.

Reply:
I agree with Max, but I also must admit, that I have problems to understand what should have traits and what not.
I have no experience with such a library and follow basically blindly.

However, I have the impression that such a library might be more suitable for projects, with a huge user base and awesome devs. Like ipython.

Reply:
I don't really see your point against having it as a traitlet. Perhaps technically it is not a configuration, but it definitely makes using EventFileReader more efficient to use if the trait is defined in the class instead of the tool. In what situation are you going to use an EventFileReader without having to then create a `input_path` trait for the tool, every single time? It just seems like extra unnecessary work we put on the user...

If we do remove it as a traitlet, do you agree that the "default path" functionality could be kept in the `__init__`?

```python
if not input_path:
   input_path = get_dataset('gamma_test.simtel.gz')
```

Reply:
Advantages of having it as a trait in the class:
* Disallows empty imput (`allow_none=False`)
* Automatically handles a default path
* User does not have to define a trait in the Tool each time they wish to pass a path from the command line, a functionality that majority of Tools will require.
* User can still override the traitlet configuration by passing the input_path in the class aruments (`reader = HessioFileReader(None, None, input_path="path/to/file")`)

Disadvantages of having it as a trait in the class:
* If there are multiple instances of the same reader in the script, every instance of the reader will have the same `input_path` set. However, a way around this would be to define the different input_path traitlets inside the tool and pass the paths individually to the readers in their initialisation arguments (see above). This is the exact same process that you would have to do if the trait didn't exist in the class.

Therefore I see removing the traitlet from the class as only removing functionality, and not achieving anything useful. I'd like to hear your arguments against having the trait in the class, and what it achieves.